gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -731,6 +871,34 @@ public void flush(String dbName, String tableName) throws IOException {
     }
   }
 
+  /**
+   * For a sorted collection of timestamps greater than an existitng watermark, check audit counts for completeness between
+   * a source and reference tier with a granularit if 1 hour
+   * If the audit count matches update the watermark to the timestamp
+   * @param table
+   * @param timestamps
+   * @param previousWatermark
+   * @return updated completion watermark
+   */
+  private long computeCompletenessWatermark(String table, Collection<Long> timestamps, long previousWatermark) {
+    log.info(String.format("Compute completion watermark for %s and timestamps %s with previous watermark %s", table, timestamps, previousWatermark));
+    long completionWatermark = previousWatermark;
+    try {
+      for(long timestamp : timestamps) {
+        if (timestamp > previousWatermark) {
+          if(auditCountVerifier.get().isComplete(table, timestamp, timestamp + TimeUnit.HOURS.toMillis(1))) {
```

Comment:
Let's not assume hourly completeness checks and make the window granularity configurable with a default of 1 hour.